stdlib/http/src/main/ballerina/src/http/cookie/csv_persistent_cookie_handler.bal

```diff
@@ -0,0 +1,254 @@
+    }
+}
+
+function closeReadableCSVChannel(io:ReadableCSVChannel csvChannel) {
+    var result = csvChannel.close();
+    if (result is error) {
+        log:printError("Error occurred while closing the channel: ", err = result);
+    }
+}
+
+// Updates the table with new cookie.
+function addNewCookieToTable(table<myCookie> cookiesTable, Cookie cookieToAdd) returns table<myCookie> | error {
+    table<myCookie> tableToReturn = cookiesTable;
+    var name = cookieToAdd.name;
+    var value = cookieToAdd.value;
+    var domain = cookieToAdd.domain;
+    var path = cookieToAdd.path;
+    var expires = cookieToAdd.expires;
+    var createdTime = time:format(cookieToAdd.createdTime, "yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    var lastAccessedTime = time:format(cookieToAdd.lastAccessedTime, "yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+    if (name is string && value is string && domain is string && path is string && createdTime is string && lastAccessedTime is string) {
+        myCookie c1 = { name: name, value: value, domain: domain, path: path, expires: expires is string ? expires : "-", maxAge: cookieToAdd.maxAge, httpOnly: cookieToAdd.httpOnly, secure: cookieToAdd.secure, createdTime: createdTime, lastAccessedTime: lastAccessedTime, hostOnly: cookieToAdd.hostOnly };
+        var result = tableToReturn.add(c1);
+        if (result is error) {
+            return result;
+        }
+        return tableToReturn;
+    }
+    CookieHandlingError err = error(COOKIE_HANDLING_ERROR, message = "Invalid data types for cookie attributes");
+    return err;
+}
+
+// Writes the updated table to file.
+function writeToFile(table<myCookie> cookiesTable, string fileName) returns @tainted error? {
+    io:WritableCSVChannel wCsvChannel2 = check io:openWritableCsvFile(fileName);
+    foreach var entry in cookiesTable {
+        string[] rec = [entry.name, entry.value, entry.domain, entry.path, entry.expires, entry.maxAge.toString(), entry.httpOnly.toString(), entry.secure.toString(), entry.createdTime, entry.lastAccessedTime, entry.hostOnly.toString()];
+        var writeResult = writeDataToCSVChannel(wCsvChannel2, rec);
+        if (writeResult is error) {
+            return writeResult;
+        }
+    }
+    closeWritableCSVChannel(wCsvChannel2);
+}
+
+function writeDataToCSVChannel(io:WritableCSVChannel csvChannel, string[]... data) returns error?{
+    foreach var rec in data {
+        var returnedVal = csvChannel.write(rec);
+        if (returnedVal is error) {
+            return returnedVal;
+        }
+    }
+}
+
+function closeWritableCSVChannel(io:WritableCSVChannel csvChannel) {
+    var result = csvChannel.close();
+    if (result is error) {
+        log:printError("Error occurred while closing the channel: ", err = result);
+    }
+}
+
+function checkRemoveCriteria(myCookie rec) returns boolean {
+    if (rec.name == cookieNameToRemove && rec.domain == cookieDomainToRemove && rec.path == cookiePathToRemove) {
+        return true;
+    }
+    return false;
```

Comment:
Shall we simplify this as follows?

`return rec.name == cookieNameToRemove && rec.domain == cookieDomainToRemove && rec.path == cookiePathToRemove;`